SignalServiceKit/src/Devices/OWSRecordTranscriptJob.m

```diff
@@ -78,18 +95,17 @@ - (void)runWithAttachmentHandler:(void (^)(NSArray<TSAttachmentStream *> *attach
         return;
     }
 
-    OWSAttachmentsProcessor *attachmentsProcessor =
-        [[OWSAttachmentsProcessor alloc] initWithAttachmentProtos:transcript.attachmentPointerProtos
-                                                   networkManager:self.networkManager
-                                                      transaction:transaction];
-
+    NSArray<TSAttachmentPointer *> *attachmentPointers =
+        [self.attachmentDownloads saveAttachmentPointersForAttachmentProtos:transcript.attachmentPointerProtos
```

Comment:
I've pulled out `saveAttachmentPointersForAttachmentProtos:transcript` into a separate method so that we can be explicit about doing this work separately from the downloads.

Reply:
Right, as you mentioned, this is going to conflict with #4003.

Reply:
These conflicts have been resolved.  I found some issues with `OWSMessageManager` wherein some parts of the code have to be restructured a bit.